### PR TITLE
Improvised pistol and armor. 

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
The makeshift pistol is an item which is made from several components such as the floor painter, it is worse than normal pistols like the commander and it uses budget magazines™ that can be printed from hacked autolathes also the magazine only has 4 rounds. 
ZipPistol4

The scrap armor can be crafted from cable coil and metal. Except for melee its useless against anything else.
ScrapArmor

## Why It's Good For The Game
More items are needed for scavengers and for low to mid tier ships, since just having the improvised shotgun isn't enough. 

## Changelog
:cl:
add: Added makeshift pistol that can be crafted
add: Added makeshift armor that can be crafted
/:cl:

### CODE REVIEW

File: whitesands/code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -152,3 +152,19 @@
 	mag_display = TRUE
 	weapon_weight = WEAPON_LIGHT
 	fire_rate = 4
+
```
Comment (martinlyra) [inline whitesands/code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion

```

**Review by martinlyra — CHANGES_REQUESTED** [review]

**Review by martinlyra — APPROVED** [review]

**Comment by Sparrowbird611** [comment]
I wasn't gonna comment but I can't help myself now. Pistol looks cool and is a nice idea, but the armor bugs me on so many levels, for 2 main reasons.

5% protection is so unbelievably worthless that calling this armor is an insult to actual armor, especially given this only covers the chest. Even worse given you made it easier to obtain firearms, which are generally the preferred method of combat unless you're a chad.

More importantly, most ships I've played on while I was here had at least two miners suits onboard, which are generally more useful given all the space exploration, have full body coverage, and their primary defensive niche is quite literally melee protection, at 30%. Also pretty sure they are upgradable with goliath body pieces to become even more resistant to melee.

If you want to make it qualify as lower tier armor, put it at like 20% melee resistance, and give it a little love and make it slightly resist projectiles. It might see some genuine use on asteroids.

Also slightly related to this, but I could have sworn we did have makeshift armor already with a similar crafting recipe, but it needed a hazard vest as well.

**Comment by ghost** [comment]
Yeah fair point, i agree with you. My reasoning for such armor was to have something that is more meant for looks than actual armor, and that look is well junk, moreover my reasoning for firearms was making it gimped like zip guns on goon which are easy to make but they are really crap.

**Comment by Sparrowbird611** [comment]
> Yeah fair point, i agree with you. My reasoning for such armor was to have something that is more meant for looks than actual armor, and that look is well junk, moreover my reasoning for firearms was making it gimped like zip guns on goon which are easy to make but they are really crap.

Totally fair, but if it's mostly cosmetic I don't think you should call it armor.

I'm not sure what you mean by a gimped firearm in this case, this isn't a gimped handgun. If you want to make it gimped, have every shot go wild by 90 degrees :)

**Comment by ghost** [comment]
> > Yeah fair point, i agree with you. My reasoning for such armor was to have something that is more meant for looks than actual armor, and that look is well junk, moreover my reasoning for firearms was making it gimped like zip guns on goon which are easy to make but they are really crap.
> 
> Totally fair, but if it's mostly cosmetic I don't think you should call it armor.
> 
> I'm not sure what you mean by a gimped firearm in this case, this isn't a gimped handgun. If you want to make it gimped, have every shot go wild by 90 degrees :)

Fair point, i mean my thing about making it gimped was making it have less rounds than normal pistols although thinking about it now, 8 is maybe too much. I should have lowered it down to 4. Since like for example the zip gun on goonstation only has 2 rounds it can fire, also about the armor i dunno i like the option of making purposely awful armor.

**Comment by TheNeoGamer42** [comment]
could make it have a slow firerate

or worse, make it act like a shotgun and flavor it as a gun that you need to manually cycle

**Comment by Sparrowbird611** [comment]
> > > Yeah fair point, i agree with you. My reasoning for such armor was to have something that is more meant for looks than actual armor, and that look is well junk, moreover my reasoning for firearms was making it gimped like zip guns on goon which are easy to make but they are really crap.
> > 
> > 
> > Totally fair, but if it's mostly cosmetic I don't think you should call it armor.
> > I'm not sure what you mean by a gimped firearm in this case, this isn't a gimped handgun. If you want to make it gimped, have every shot go wild by 90 degrees :)
> 
> Fair point, i mean my thing about making it gimped was making it have less rounds than normal pistols although thinking about it now, 8 is maybe too much. I should have lowered it down to 4. Since like for example the zip gun on goonstation only has 2 rounds it can fire, also about the armor i dunno i like the option of making purposely awful armor.

I want to say give it six and have it be a makeshift revolver then leave it at that.

**Comment by ghost** [comment]
Sure that sounds like a good idea, you know what 4 rounds is probably better.

**Comment by Sparrowbird611** [comment]
> Sure that sounds like a good idea, you know what 4 rounds is probably better.

Up to you. I'd put it at a solid five, so you can realistically kill an unarmored person if you land every shot, which is basically mythical.

**Review by MarkSuckerberg — APPROVED** [review]

File: whitesands/code/modules/projectiles/guns/ballistic/automatic.dm [hunk 2]
```diff
@@ -151,3 +151,18 @@
 	mag_display = TRUE
 	weapon_weight = WEAPON_LIGHT
 	fire_rate = 4
+/obj/item/gun/ballistic/automatic/ZipPistol
```
Comment (ZephyrTFA) [inline whitesands/code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion

/obj/item/gun/ballistic/automatic/ZipPistol
```

**Comment by ghost** [comment]
Please i beg you github eldritch beings let the checks all be successful...

File: whitesands/code/datums/components/crafting/recipes.dm [hunk 3]
```diff
@@ -145,3 +145,25 @@
...
+	tools = list(TOOL_SCREWDRIVER)
+	time = 100
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/ScrapArmor
+	name = "Scrap Armor"
+	result = /obj/item/clothing/suit/armor/vest/scraparmor
+	time = 60
+	reqs = list(/obj/item/stack/sheet/metal = 10,
```
Comment (ZephyrTFA) [inline whitesands/code/datums/components/crafting/recipes.dm]:
```suggestion
	reqs = list(
		/obj/item/stack/sheet/metal = 10,
```

File: whitesands/code/datums/components/crafting/recipes.dm [hunk 4]
```diff
@@ -145,3 +145,25 @@
...
+				/obj/item/stack/cable_coil = 20)
```
Comment (ZephyrTFA) [inline whitesands/code/datums/components/crafting/recipes.dm]:
```suggestion
		/obj/item/stack/cable_coil = 20,
	)
```

File: whitesands/code/modules/clothing/suits/armor.dm [hunk 5]
```diff
@@ -32,3 +32,11 @@
...
 	item_state = "syndiearmor"
 
+/obj/item/clothing/suit/armor/vest/scraparmor
+	name = "\improper Scrap Armor"
+	desc = "Who thought this is a good idea for armor?"
+	icon = 'whitesands/icons/obj/clothing/suits.dmi'
+	mob_overlay_icon = 'whitesands/icons/mob/clothing/suits.dmi'
+	icon_state = "scraparmor"
+	item_state = "scraparmor"
+	armor = list(melee = 5, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)
```
Comment (ZephyrTFA) [inline whitesands/code/modules/clothing/suits/armor.dm]:
```suggestion
	armor = list(melee = 5)
```

File: whitesands/code/datums/components/crafting/recipes.dm [hunk 6]
```diff
@@ -145,3 +145,25 @@
 	time = 20
 	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
 	category = CAT_MISC
+
+/datum/crafting_recipe/ZipPistol
```
Comment (ZephyrTFA) [inline whitesands/code/datums/components/crafting/recipes.dm]: Change these names to snake_case please

File: whitesands/code/datums/components/crafting/recipes.dm [hunk 7]
```diff
@@ -145,3 +145,25 @@
...
+			/obj/item/stack/cable_coil = 15,
+			/obj/item/weaponcrafting/receiver = 1,
+			/obj/item/floor_painter = 1,
+			/obj/item/stack/packageWrap = 10)
+	tools = list(TOOL_SCREWDRIVER)
+	time = 100
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/ScrapArmor
```
Comment (ZephyrTFA) [inline whitesands/code/datums/components/crafting/recipes.dm]: See above

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: whitesands/code/datums/components/crafting/recipes.dm (hunk 6 above)
Comment (ghost) [reply whitesands/code/datums/components/crafting/recipes.dm]: I think its resolved now, emphasis on the word think.

File: whitesands/code/datums/components/crafting/recipes.dm [hunk 8]
```diff
@@ -145,3 +145,27 @@
 	time = 20
 	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
 	category = CAT_MISC
+
+/datum/crafting_recipe/Zip_Pistol
```
Comment (ZephyrTFA) [inline whitesands/code/datums/components/crafting/recipes.dm]:
```suggestion
/datum/crafting_recipe/zip_pistol
```

File: whitesands/code/datums/components/crafting/recipes.dm [hunk 9]
```diff
@@ -145,3 +145,27 @@
...
+			/obj/item/stack/cable_coil = 15,
+			/obj/item/weaponcrafting/receiver = 1,
+			/obj/item/floor_painter = 1,
+			/obj/item/stack/packageWrap = 10)
+	tools = list(TOOL_SCREWDRIVER)
+	time = 100
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/Scrap_Armor
```
Comment (ZephyrTFA) [inline whitesands/code/datums/components/crafting/recipes.dm]:
```suggestion
/datum/crafting_recipe/scrap_armor
```

File: whitesands/code/modules/clothing/suits/armor.dm [hunk 10]
```diff
@@ -32,3 +32,11 @@
 	icon_state = "syndiearmor"
 	item_state = "syndiearmor"
 
+/obj/item/clothing/suit/armor/vest/scraparmor
```
Comment (ZephyrTFA) [inline whitesands/code/modules/clothing/suits/armor.dm]:
```suggestion
/obj/item/clothing/suit/armor/vest/scrap_armor
```

File: whitesands/code/modules/projectiles/boxes_magazines/ammo_boxes.dm [hunk 11]
```diff
@@ -2,3 +2,12 @@
 	name = "ammo box (9mm rubbershot)"
 	desc = "Contains less-than-lethal ammunition."
 	ammo_type = /obj/item/ammo_casing/c9mm/rubbershot
+
+/obj/item/ammo_box/magazine/ZipAmmo9mm
```
Comment (ZephyrTFA) [inline whitesands/code/modules/projectiles/boxes_magazines/ammo_boxes.dm]:
```suggestion
/obj/item/ammo_box/magazine/zip_ammo_9mm
```

File: whitesands/code/modules/projectiles/guns/ballistic/automatic.dm [hunk 12]
```diff
@@ -151,3 +151,19 @@
 	mag_display = TRUE
 	weapon_weight = WEAPON_LIGHT
 	fire_rate = 4
+
+/obj/item/gun/ballistic/automatic/ZipPistol
```
Comment (ZephyrTFA) [inline whitesands/code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
/obj/item/gun/ballistic/automatic/zip_pistol
```

File: whitesands/code/modules/research/designs/autolathe_designs.dm [hunk 13]
```diff
@@ -7,3 +7,11 @@
 	build_path = /obj/item/floor_painter
 	category = list("initial","Tools","Tool Designs")
 	departmental_flags = DEPARTMENTAL_FLAG_ENGINEERING | DEPARTMENTAL_FLAG_SERVICE
+
+/datum/design/ZipAmmo9mm
```
Comment (ZephyrTFA) [inline whitesands/code/modules/research/designs/autolathe_designs.dm]:
```suggestion
/datum/design/zip_ammo_9mm
```

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

**Review by MarkSuckerberg — APPROVED** [review]

**Review by ZephyrTFA — APPROVED** [review]

**Comment by shiptest-server** [comment]
#### Test Merge Deployed

##### Server Instance
shiptest

##### Revision
Origin: ca83a63d9154e7bd47271dd9f5ce5d44eb19d7af
Pull Request: 4893edffa2c94e1feb50343a4eb33e0b5c331bd0
Server: b9679cc28f47ef4054511559ac3e5f35dd3100bb

**Comment by shiptest-server** [comment]
#### Test Merge Updated

##### Server Instance
shiptest

##### Revision
Origin: ca83a63d9154e7bd47271dd9f5ce5d44eb19d7af
Pull Request: 4893edffa2c94e1feb50343a4eb33e0b5c331bd0
Server: 52f18b307da9ed9bafb2f173c5db9806fde13a70

**Pull request merged**